### PR TITLE
[ISSUE #2931]Resolve hystrix plugin configuration rules do not take effect

### PR DESCRIPTION
**Pull request by zouchangfu**
Resolve hystrix plugin configuration rules do not take effect

### CODE REVIEW

**Review by JooKS-me — COMMENTED** [review]
Pls check ci, there are some Checkstyle error

**Comment by erdengk** [comment]
@zouchangfu 
I made the same changes to the project as you. 
But it still can't solve the problem of hystrix plugin. At present (v2.4.2), hystrix can capture requests, but cannot implement a circuit breaker mechanism.

**Comment by zouchangfu** [comment]
I tested that it works and modified it in my company

**Comment by erdengk** [comment]
hi，@zouchangfu 
I want to implement the circuit breaker mechanism of hystrix through configuration, can you guide me how to configure it?

**Comment by zouchangfu** [comment]
@erdengk What problems have you encountered, you can list your problems.

**Comment by erdengk** [comment]
hi, @zouchangfu 
first of all thank you for your willingness to help me.

I want to write integration tests for hystrix. #2985
So I need to implement hystrix's circuit breaker.
For example, after the system reaches the maximum number of requests, hystrix performs fuse.
The system will return to the user a ""You have been restricted, please try again later!"" prompt.

For example in this file,
`shenyu-integrated-test/shenyu-integrated-test-http/src/test/java/org/apache/shenyu/integrated/test/http/combination/SentinelPluginTest.java`

```java
    @Test
    public void test() throws IOException {
        String selectorAndRulesResult =
                initSelectorAndRules(PluginEnum.SENTINEL.getName(), "", buildSelectorConditionList(), buildRuleLocalDataList(null));
        assertThat(selectorAndRulesResult, is("success"));

        Type returnType = new TypeToken<Map<String, Object>>() {
        }.getType();
        Map<String, Object> result = HttpHelper.INSTANCE.postGateway(TEST_SENTINEL_PATH, returnType);
        assertNotNull(result);
        assertEquals("pass", result.get("msg"));
        result = HttpHelper.INSTANCE.postGateway(TEST_SENTINEL_PATH, returnType);
        assertEquals("You have been restricted, please try again later!", result.get("message"));
    }

```
SentinelPlugin implements a circuit breaker mechanism. The first request will return pass, and then the system will be blown, and a prompt of ""You have been restricted, please try again later!" will be returned.

I would like to ask you to give an example, how do I change the SelectorList and rule of the hystrix plugin to achieve this circuit breaker mechanism?

**Comment by zouchangfu** [comment]
@erdengk You need to set the commandKey and groupKey or use the contextPath plugin to configure the contextPath
image
image

File: shenyu-plugin/shenyu-plugin-hystrix/src/main/java/org/apache/shenyu/plugin/hystrix/handler/HystrixPluginDataHandler.java [hunk 1]
```diff
@@ -39,10 +41,17 @@
 
     @Override
     public void handlerRule(final RuleData ruleData) {
-        HystrixPropertiesFactory.reset();
         Optional.ofNullable(ruleData.getHandle()).ifPresent(rule -> {
             HystrixHandle hystrixHandle = GsonUtils.getInstance().fromJson(rule, HystrixHandle.class);
-            CACHED_HANDLE.get().cachedHandle(CacheKeyUtils.INST.getKey(ruleData), hystrixHandle);
+            String key = CacheKeyUtils.INST.getKey(ruleData);
+            Optional.ofNullable(CACHED_HANDLE.get().obtainHandle(key)).ifPresent(hystrixHandleCache -> {
+                if (hystrixHandleCache.getMaxConcurrentRequests() != hystrixHandle.getMaxConcurrentRequests()) {
```
Comment (yu199195) [inline shenyu-plugin/shenyu-plugin-hystrix/src/main/java/org/apache/shenyu/plugin/hystrix/handler/HystrixPluginDataHandler.java:48]: only MaxConcurrentRequests change， and reset？

**Comment by zouchangfu** [comment]
@yu199195 I retain the original reset method, currently I test to MaxConcurrentRequests modification did not take effect, so I modify this bug

**Comment by erdengk** [comment]
@zouchangfu 
Thank you for your reply.
Can you give an example? About how to set commandKey and groupKey

**Comment by zouchangfu** [comment]
@erdengk you can see this https://shenyu.apache.org/docs/plugin-center/fault-tolerance/hystrix-plugin/
image

**Comment by erdengk** [comment]
@zouchangfu 
ok，thanks

**Review by yu199195 — APPROVED** [review]

**Pull request merged**